sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1102,13 +1097,63 @@ def get_cmis_dp_init_duration_secs(self, api):
     def get_cmis_dp_deinit_duration_secs(self, api):
         return api.get_datapath_deinit_duration()/1000
 
-    def is_cmis_application_update_required(self, api, channel, speed):
+    def get_cmis_host_lanes(self, api, host_lane_count, channel, speed):
+        """
+        Retrieves bitmask of host lanes based on host lane count, channel and speed
+
+        Args:
+            api:
+                XcvrApi object
+            host_lane_count:
+                Integer, number of lanes on the host side
+            channel:
+                Integer, channel id of the group which the host lanes belong to (1 based)
+            speed:
+                Integer, the port speed of the host interface
+
+        Returns:
+            Integer, a bitmask of the lanes on the host side
+            e.g. 0x5 for lane 0 and lane 2.
+        """
+        host_lanes = 0
+
+        if channel > host_lane_count:
+            self.log_error("Failed to get host lanes as channel > host_lane_count!")
+            return host_lanes
+
+        if channel != 0:
+            appl = self.get_cmis_application_desired(api, host_lane_count, speed)
+            if appl < 1:
+                self.log_error("Failed to get host lanes as no suitable app for the port")
+                return host_lanes
+
+            host_lane_assignment_option = api.get_host_lane_assignment_option(appl)
+            curr_channel_num = 0
+            for bit in range(8):
+                mask = 1 << bit
+                if host_lane_assignment_option & mask != 0:
+                    curr_channel_num += 1
+                    if curr_channel_num == channel:
+                        for i in range(host_lane_count):
+                            host_lanes |= mask
+                            mask = mask << 1
+                        break
+                mask = mask << 1
```

Comment:
this mask value is overwritten at the line 1133, so whats the purpose of updating here?